### DISCUSSION TITLE
Best way to save lexical state for scalability

### DISCUSSION THREAD

**Discussion by ChanManChan** [facebook/lexical#4878] [Q&A] [answered]
![image](https://github.com/facebook/lexical/assets/42992965/71183e5f-cd04-4155-b50c-63317c65b237)

Initially I tried saving the editor state as JSON in the DB but to render it I had to use lexical editor again which is very bad for performance. So I decided to save the HTML instead and just by keeping the lexical custom css outside the lexical module I can retain the styles and show the desired output without the heavy weight editor. But if we were to expand the functionality to support more advanced features like having a poll component, the generated HTML output becomes unusable outside the lexical context.

Is there a way to generate the 2. HTML (_refer the attached picture_ - The HTML that gets generated inside the lexical editor) from 1. HTML (_refer the attached picture_ - The HTML that gets generated by $generateHtmlFromNodes(editor, null) which is then saved to the DB)?

I need to show the poll component and other custom components outside of the lexical environment (so that the page can be lightweight)

PS:- I think I might know the answer, it could be related to correctly configuring the exportDOM function to meet my needs right?
`  exportDOM() {
    const element = document.createElement('span')
    element.setAttribute('data-lexical-poll-question', this.__question)
    element.setAttribute('data-lexical-poll-options', JSON.stringify(this.__options))
    return { element }
  }`

But I wanted to know if there are any easier ways of doing this.

Thank you for taking the time to read this.

**Comment by acywatson** [comment, marked as answer]
> it could be related to correctly configuring the exportDOM function to meet my needs

Yes, that's right. You can export literally any HTML you want with that function.

> which is very bad for performance

Would be very interested to hear specific performance problems you encountered doing this.

**Reply by ChanManChan** [reply]
The issue is with the extra scripts that the app had to download just to show the list view. I wanted to keep the home page as light as possible. That is why I'm lazily loading lexical editor only when it is absolutely required. Therefore I needed an alternative solution.

Thanks for you quick response, much appreciated :)
